gui/gui/configpanel/local_file_picker.py

```diff
@@ -0,0 +1,82 @@
+import platform
+from pathlib import Path
+from typing import Optional
+
+from nicegui import events, ui
+
+
+class local_file_picker(ui.dialog):
+
+    def __init__(self, directory: str, *,
+                 upper_limit: Optional[str] = ..., multiple: bool = False, show_hidden_files: bool = False) -> None:
+        """Local File Picker
+
+        This is a simple file picker that allows you to select a file from the local filesystem where NiceGUI is running.
+
+        :param directory: The directory to start in.
+        :param upper_limit: The directory to stop at (None: no limit, default: same as the starting directory).
+        :param multiple: Whether to allow multiple files to be selected.
+        :param show_hidden_files: Whether to show hidden files.
+        """
+        super().__init__()
+
+        self.path = Path(directory).expanduser()
+        if upper_limit is None:
+            self.upper_limit = None
+        else:
+            self.upper_limit = Path(directory if upper_limit == ... else upper_limit).expanduser()
+        self.show_hidden_files = show_hidden_files
+
+        with self, ui.card():
+            self.add_drives_toggle()
+            self.grid = ui.aggrid({
+                'columnDefs': [{'field': 'name', 'headerName': 'File'}],
+                'rowSelection': 'multiple' if multiple else 'single',
+            }, html_columns=[0]).classes('w-96').on('cellDoubleClicked', self.handle_double_click)
+            with ui.row().classes('w-full justify-end'):
+                ui.button('Cancel', on_click=self.close).props('outline')
+                ui.button('Ok', on_click=self._handle_ok)
+        self.update_grid()
+
+    def add_drives_toggle(self):
+        if platform.system() == 'Windows':
+            import win32api
+            drives = win32api.GetLogicalDriveStrings().split('\000')[:-1]
+            self.drives_toggle = ui.toggle(drives, value=drives[0], on_change=self.update_drive)
```

Comment:
Can some os agnostic lib be used to handle files, like os or pathlib?

Reply:
I've barely even looked at this file to be honest. I took it straight from [NiceGUI examples ](https://github.com/zauberzeug/nicegui/blob/main/examples/local_file_picker/local_file_picker.py)(not sure why this isn't added in the main library). I just tried if it worked as intended and didn't bother more with it. I also have no idea what's compatible with Windows and not.